src/hotspot/cpu/aarch64/globals_aarch64.hpp

```diff
@@ -110,7 +110,10 @@ define_pd_global(intx, InlineSmallCode,          1000);
   product(int, SoftwarePrefetchHintDistance, -1,                        \
           "Use prfm hint with specified distance in compiled code."     \
           "Value -1 means off.")                                        \
-          range(-1, 4096)
+          range(-1, 4096)                                               \
+  product(ccstr, OnSpinWaitImpl, "none",                                \
+          "Use instructions to implement java.lang.Thread.onSpinWait()."\
+          "Options: none, Nnop, Nisb, Nyield, where optional N is 2..9.")
```

Comment:
Could this N be changed to 2..99 instead of 2..9? I tested this with SpinWaitBench.pong:totalSpins (ops/us), 9nop or 9yield is unable to provide a similar pause time as 1isb, maybe ~20 can do. A larger range can be more convenient for future experiments/tunings. Thanks.

Reply:
As Andrew suggested to have separated options for the name and the count, this won't be an issue.

Reply:
It was more of a question than a suggestion. I'm not sure, and I'd like you to have a look at existing options to see if there's a precedent.

Reply:
I like the suggestion to have them separated. It makes code simpler. A separate `OnSpinWaitInstCount` uses standard ways to define it as an integer with a range. I don't need to write code for this. Problems with the combined approach start when a number of instructions to use needs to be more than 9.

I have found no precedent of combined options.